README.md

```diff
@@ -59,40 +60,95 @@ Idempotent consumer group creation:
+                           group
+                           consumer
+                           #(println "Yum yum, tasty message" %)
+                           opts))
 ```
 
-Consumer behaviour is as follows:
+Consumer behaviour when there is only one stream is as follows:
 
  - Calls the callback for every message received, with the message
    coerced into a keywordized map, and acks the message.
    If the callback throws an exception the message will not be acked
  - Processes all pending messages on startup before processing new ones
  - Processes new messages until either:
    - The consumer is unblocked (see `unblock-consumers!`)
-   - There are no messages delivered during the time it was blocked waiting
-     for a new message, upon which it will check for pending messages and
-     begin processing the backlog if any are found, returning to wait for
-     new messages when the backlog is cleared
+   - There are no messages delivered during the time it was blocked
+     waiting for a new message. If this happens, it will check for
+     pending messages and begin processing the backlog if any are
+     found, returning to wait for new messages when the backlog is
+     cleared.
+
+
+When checking for pending messages, if it has been sufficiently long
+since the last check, it will check for idle messages on the backlog
+of other consumers and claim them, or putting messages on the dlq if
+they have been retried too many times. This ensures that even if a
+consumer dies, its messages will still be processed.
+
+A consumer can also be passed multiple streams:
+
+```clj
+(def opts {:block 5000
+           :control-fn cs/default-control-fn})
+
+(future
+ (cs/start-multi-consumer! conn-opts
+                           [stream1 stream2 stream3]
+                           group
+                           consumer
+                           #(println "Yum yum, tasty message" %)
+                           opts))
+```
+
+When passed multiple streams, the consumer will behave similarly to
+when it is passed a single stream, except it will process messages
+from the first stream, then the second stream, then the third, etc. If
+a new message arrives on a higher priority stream while it is
+receiving messages on a lower priority stream, it will process the
+higher priority message as soon as it has finished processing its
+current message.
+
+Options to the consumer consist of:
+- `:block` ms to block waiting for a new message when there are no
+  pending messages on any of the streams
+- `:control-fn` a function for controlling the flow of operation, see `default-control-fn`
+- `:claim-opts` an options map for configuring how messages are
+  claimed from other consumers. See [Recovering from failures](#recovering-from-failures) for available options.
 
- Options to the consumer consist of:
+The behaviour of a single stream of a consumer that has multiple streams is given (slightly simplified) by this flowchart:
```

Comment:
"a single stream of a consumer"? Don't you just mean "The behaviour of a consumer that has multiple streams..."?

Reply:
I mean that the "current state" (the two ellipses at the top of the diagram) represents the state as-recorded for a single stream. It isn't always looks at pending messages for all streams or looking at new messages for all streams, but can be a mixture of the two. Not sure how to make that clearer though...

Reply:
How about "Each stream being processed by a multi-stream consumer will be processed as shown in this flowchart"?